src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -86,34 +86,64 @@ internal partial class Binder
         /// <summary>
         /// Bind and return a single type parameter constraint clause.
         /// </summary>
-        private TypeParameterConstraintClause BindTypeParameterConstraints(string name, SeparatedSyntaxList<TypeParameterConstraintSyntax> constraintsSyntax, DiagnosticBag diagnostics)
+        private TypeParameterConstraintClause BindTypeParameterConstraints(
+            string name, SeparatedSyntaxList<TypeParameterConstraintSyntax> constraintsSyntax, DiagnosticBag diagnostics)
         {
             var constraints = TypeParameterConstraintKind.None;
             var constraintTypes = ArrayBuilder<TypeSymbol>.GetInstance();
+            var isStruct = false;
 
-            foreach (var syntax in constraintsSyntax)
+            for (int i = 0, n = constraintsSyntax.Count; i < n; i++)
             {
+                var syntax = constraintsSyntax[i];
                 switch (syntax.Kind())
                 {
                     case SyntaxKind.ClassConstraint:
+                        if (i != 0)
+                        {
+                            diagnostics.Add(ErrorCode.ERR_RefValBoundMustBeFirst, syntax.GetFirstToken().GetLocation());
+                        }
+
                         constraints |= TypeParameterConstraintKind.ReferenceType;
-                        break;
```

Comment:
to me 'break' indicates 'there is more code to run after this.  jump from where i am to that point so that all the remainder code can run'.  'continue' means "i'm done, go process the next element".  With continue i can know nothing more will run.  With 'break' i have to go validate that no one has added code that will run after me.  'continue' seems much clearer.  It's also much clearer to have uniform code flow in this section.  IIRC, we had a mix of both types.  I'd much refer one single type for consistency.